pkg/operator/storageclasscontroller/vmware.go

```diff
@@ -54,3 +324,24 @@ func (c *StorageClassController) getCredentials(cfg *vsphere.VSphereConfig) (str
 
 	return string(username), string(password), nil
 }
+
+func notFoundError(err error) bool {
+	errorString := err.Error()
+	r := regexp.MustCompile("404")
```

Comment:
This looks error prone. How does the message look like? Can we at least grep for `^404`?

Reply:
This is how message looks like btw:

```
func (e *statusError) Error() string {
	return fmt.Sprintf("%s %s: %s", e.res.Request.Method, e.res.Request.URL, e.res.Status)
}
```

Unfortunately vsphere SDK has chosen to obscure error code from users of the library here. It returns following type as an error:

```
type statusError struct {
	res *http.Response
}
```

since only public method that struct has is `Error()` - I can't extract the error code.

Reply:
So `404` string will be at the end - may be I can look for `404` towards the end of the string. Not sure if will be that much better though. :(

Reply:
actually - since it prints `Status` and not `StatusCode` - it is even more crazier, because that would print `blah blah - 404 Not Found`